Source/GTMSessionFetcher.h

```diff
@@ -608,6 +611,22 @@ NSData *_Nullable GTMDataFromInputStream(NSInputStream *inputStream, NSError **o
 }  // extern "C"
 #endif
 
+// Adds HTTP header(s) to a request before it's sent out. See `-[GTMSessionFetcherService
+// addHeaderDecorator:]` and `-[GTMSessionFetcherService removeHeaderDecorator:]`.
+@protocol GTMFetcherHeaderDecoratorProtocol <NSObject>
+
+// Return YES if this request should have its headers decorated, NO otherwise.
+- (BOOL)shouldDecorateHeadersForRequest:(NSURLRequest *)request;
```

Comment:
We should probably comment queue/threading expectations for these methods. For example, neither should make I/O calls that could block, since both are potentially called from the main queue.

Reply:
Thanks, added documentation on threading contract.